rfcs/72-editor.md

```diff
@@ -0,0 +1,508 @@
+
+Assets can be added to your project through the `bevy` tool:
+
+```
+bevy assets add <asset-name>
+```
+
+Likewise, you can remove the asset with:
+
+```
+bevy assets remove <asset-name>
+```
+
+In general, assets come in three varieties: *code*, *data*, and *template*. Code
+assets are little more than Cargo crates, many can even be installed with the
+`cargo` command.
+
+Data assets are overlaid onto the existing project, and usually go into the
+`assets` directory. Frequently these assets are large, should not be added to
+the repository directly, or should be loaded at build time from an external
+source. Therefore, in addition to copying the assets into the project, data
+assets can also be references to an asset at an external location. **Note that
+while referenced asses can be removed from the project, copied data assets can
+not!**
+
+Finally, template assets are not added to a project, but are instead used by the
+`bevy` tool to start new projects. The `-t` flag is used to pick a project
+template:
+
+```
+bevy new -t <template> <project>
+```
+
+For example, we can instead create our Flapper project using the `foxtrot`
+template:
+
+```
+$ bevy new -t foxtrot flapper
+```
+
+Sometimes it makes sense to install an asset onto a system rather then adding it
+to a project. The most common case for this is template assets, which are used
+to create the initial project. To install an asset to the system you can use the `install` command:
+
+```
+bevy install <asset-name>
+```
+
+This will install the asset to your user's data directory. If you want to
+install an asset to a different location, you can set the `BEVY_ASSETS_DIR`
+environment variable.
+
+### Asset Repositories
+
+In addition to the main repository for assets, projects can use additional
+repositories for storing and retrieving assets. You can add a new repository with:
+
+```
+bevy repo add [repository-name] <repository-url>
+```
+
+This adds a repository to the current project. Optionally, a repository name can
+be provided, which is used by `bevy assets`:
+
+```
+bevy assets add repository-name/asset-name
+```
+
+Despite their similarities asset repos are different from git remotes in one
+crucial way: they are generally checked into the repository. In this sense, they
+are more akin to git submodules or git large file storage than pure remotes.
+
+Sometimes it makes sense to have a remote stored system wide, rather than with
+any particular project. The `-g` flag allows you to add repositories globally.
+
+```
+bevy repo add -g [repository-name] <repository-url>
+```
+
+Alternatively, you can set the `BEVY_ASSET_REPO` environment variable to one or
+more asset repositories.
+
+### Publishing Tools
+
+Bevy also helps you publish your games, either by building distributables, or
+even having CI distribute builds each time its pushed.
+
+Use `bevy build` to build your game:
+
+```
+bevy build <target>
```

Comment:
I think `target` should be optional: the default should build for Mac / Linux / Windows. And then allow configuration of default build targets.

Reply:
What about `build.rs` and any sort of preprocessing/baking.
Would this be the command that does it?

As of right now it seems like a light wrapper around `cargo build`.
Worth noting, it should allow all of the arguments `cargo build` takes.
Possibly making default `cargo build` arguments a project configuration.